test/integration_test/basic_test.go

```diff
@@ -208,3 +221,70 @@ func BasicUpgrade(tc *TestCase) func(*testing.T) {
 		require.NoError(t, err)
 	}
 }
+
+func InstallWithTelemetry(tc *TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
+
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*op_corev1.StorageCluster)
+		require.True(t, ok)
+
+		testInstallWithTelemetry(t, cluster)
+	}
+}
+
+func testInstallWithTelemetry(t *testing.T, cluster *op_corev1.StorageCluster) {
+	secret := testutil.GetExpectedSecret(t, "/specs/pure-telemetry-cert.yaml")
+	require.NotNil(t, secret)
+
+	_, err := coreops.Instance().GetSecret(component.TelemetryCertName, "kube-system")
+	require.True(t, err == nil || errors.IsNotFound(err))
+
+	if errors.IsNotFound(err) {
+		secret, err = coreops.Instance().CreateSecret(secret)
+		require.NoError(t, err)
+	} else {
+		secret, err = coreops.Instance().UpdateSecret(secret)
+		require.NoError(t, err)
+	}
+
+	// Deploy portworx with telemetry set to true
+	imageListMap, err := testutil.GetImagesFromVersionURL(pxSpecGenURL)
+	require.NoError(t, err)
+
+	err = constructStorageCluster(cluster, pxSpecGenURL, imageListMap)
+	require.NoError(t, err)
+
+	cluster.Name = "test-sc"
+	cluster.Spec.Monitoring = &op_corev1.MonitoringSpec{
+		Telemetry: &op_corev1.TelemetrySpec{
+			Enabled: true,
+		},
+	}
+
+	cluster, err = createStorageCluster(cluster)
+	require.NoError(t, err)
+
+	err = testutil.ValidateTelemetry(
```

Comment:
we could move this validation to `BasicInstall`, check if telemetry is enabled then validate it, then you should be able to just call Basic install test func and remove this whole function `testInstallWithTelemetry`